library/core/src/fmt/num_buffer.rs

```diff
@@ -0,0 +1,62 @@
+use crate::mem::MaybeUninit;
+
+/// Trait used to describe the maximum number of digits in decimal base of the implemented integer.
+#[unstable(feature = "int_format_into", issue = "138215")]
+pub trait NumBufferTrait {
+    /// Maximum number of digits in decimal base of the implemented integer.
+    const BUF_SIZE: usize;
+}
+
+macro_rules! impl_NumBufferTrait {
+    ($($signed:ident, $unsigned:ident,)*) => {
+        $(
+            #[unstable(feature = "int_format_into", issue = "138215")]
+            impl NumBufferTrait for $signed {
+                const BUF_SIZE: usize = $signed::MAX.ilog(10) as usize + 2;
+            }
+            #[unstable(feature = "int_format_into", issue = "138215")]
+            impl NumBufferTrait for $unsigned {
+                const BUF_SIZE: usize = $signed::MAX.ilog(10) as usize + 1;
+            }
+        )*
+    }
+}
+
+impl_NumBufferTrait! {
+    i8, u8,
+    i16, u16,
+    i32, u32,
+    i64, u64,
+    isize, usize,
+    i128, u128,
+}
+
+/// A buffer wrapper of which the internal size is based on the maximum
+/// number of digits the associated integer can have.
+#[unstable(feature = "int_format_into", issue = "138215")]
+#[derive(Debug)]
+pub struct NumBuffer<T: NumBufferTrait> {
+    // FIXME: Once const generics feature is working, use `T::BUF_SIZE` instead of 40.
+    pub(crate) buf: [MaybeUninit::<u8>; 40],
```

Comment:
This was a great frustration to discover that I couldn't just use `[MaybeUninit::<u8>; T::BUF_SIZE]` so until it's actually supported, they will all use the max size. :disappointed:

Reply:
@lqd provided me a way to do it so all solved now. :)

Reply:
Ah sadly it breaks something else so it cannot. For reference here is the code showing the feature that was supposed to unlock the situation: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2024&gist=11cfd87eaeeef6481be6433123fc0caf (the `min_generic_const_args` feature).